packages/@lwc/synthetic-shadow/src/polyfills/event-target/polyfill.ts

```diff
@@ -17,55 +17,62 @@ import {
 import { isHostElement } from '../../faux-shadow/shadow-root';
 import { getEventListenerWrapper } from '../../shared/event-target';
 
-function addEventListener(
+function patchedAddEventListener(
     this: EventTarget,
-    type: string,
-    listener: EventListenerOrEventListenerObject,
-    optionsOrCapture?: boolean | AddEventListenerOptions
+    _type: string,
+    _listener: EventListenerOrEventListenerObject,
+    _optionsOrCapture?: boolean | AddEventListenerOptions
 ) {
-    if (arguments.length < 2) {
-        // Delegate to the browser for throwing these errors.
+    if (isHostElement(this)) {
+        // Typescript does not like it when you treat the `arguments` object as an array
         // @ts-ignore type-mismatch
-        return nativeAddEventListener.apply(this, arguments);
+        return addCustomElementEventListener.apply(this, arguments);
```

Comment:
I don't think is right here: `addCustomElementEventListener` expects the element as the first argument and not as this value. It expects the following arguments today.

```suggestion
        return addCustomElementEventListener(this, ...arguments);
```

The `addCustomElementEventListener` does also some validation internally. Those validations should be removed.